code/modules/vending/mail.dm

```diff
@@ -0,0 +1,299 @@
+/obj/machinery/mailsorter
+	name = "mail sorter"
+	desc = "A large mail sorting unit. Sorting mail since 1987!"
+	icon = 'icons/obj/machines/mailsorter.dmi'
+	icon_state = "mailsorter"
+	layer = BELOW_OBJ_LAYER
+	density = TRUE
+	max_integrity = 300
+	integrity_failure = 0.33
+	var/light_mask = "mailsorter-light-mask"
+	var/panel_type = "panel"
+
+	circuit = /obj/item/circuitboard/machine/mailsorter
+
+	/// How much mail can the mail sorter store.
+	var/mail_capacity = 100
+	/// Bool that returns if the machine is already sorting mail.
+	var/now_sorting = FALSE
+	/// What the machine is currently doing. Can be "sorting", "idle", "yes", "no".
+	var/currentstate = "idle"
+	/// List of all mail that's inside the mailbox.
+	var/list/mail_list = list()
+	/// The direction in which the mail will be unloaded.
+	var/output_dir = SOUTH
+	/// The turf to unload mail at.
+	var/turf/unload_turf = null
```

Comment:
is this ever cleaned up?

Reply:
What exactly is supposed to be cleaned up here? I don't understand